pkg/e2e/infra/utils.go

```diff
@@ -211,3 +231,54 @@ func getScaleClient() (scale.ScalesGetter, error) {
 	}
 	return scaleClient, nil
 }
+
+func scaleWorkers(client runtimeclient.Client, replicas int) error {
```

Comment:
this is updated in later commit to `scaleAWorker`. It is intentionally high abstracted and intentionally picking a random worker machineSet to test
There's a different test story in this PR which test against specific machineSets to validate they can scale simulatanously and the cluster grows/decrease accordingly

Reply:
Are you saying it's intended to call the `scaleAWorker` two times in a row and get different machine set to be scaled?

Reply:
actually no, because that would break the cluster size expectations.
There is programatically guarantee of the ordering, unless something else manipulate the cluster nodes/machines in the middle of the calls. Which might be the case as we introduce more concurrent uses cases and parallel runs. I'll update it to be more deterministic

Reply:
> this is updated in later commit to `scaleAWorker`. It is intentionally high abstracted and intentionally picking a random worker machineSet to test

It would be better to have different tests for this then. One of the tests should be predictable.

Reply:
updated scaleAWorker to pick a known set